### PR TITLE
script: adapt build task to the recent omnibus refactoring

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

This PR replaces `agent.omnibus-build` by `omnibus.build`, following the changes in https://github.com/DataDog/datadog-agent/pull/24568

### CODE REVIEW

**Review by KSerrania — APPROVED** [review]

File: scripts/build_script.sh [hunk 1]
```diff
@@ -48,13 +48,18 @@ if [ -n "$INTEGRATIONS_CORE_REF" ]; then
     export INTEGRATIONS_CORE_VERSION="$INTEGRATIONS_CORE_REF"
 fi
 
+INVOKE_TASK="omnibus.build"
+if ! inv --list | grep -F "${INVOKE_TASK}"; then
```
Comment (CelianR) [inline scripts/build_script.sh:52]: I think you want to add the `-q` flag to grep in this case
```suggestion
if ! inv --list | grep -qF "$INVOKE_TASK"; then
```

File: scripts/build_script.sh [hunk 2]
```diff
@@ -48,13 +48,18 @@ if [ -n "$INTEGRATIONS_CORE_REF" ]; then
...
+    INVOKE_TASK="agent.omnibus-build"
+fi
+
 # Launch omnibus build
 if [ "$SIGN" = "true" ]; then
     # Unlock the keychain to get access to the signing certificates
     security unlock-keychain -p "$KEYCHAIN_PWD" "$KEYCHAIN_NAME"
-    inv -e agent.omnibus-build --hardened-runtime --python-runtimes "$PYTHON_RUNTIMES" --major-version "$AGENT_MAJOR_VERSION" --release-version "$RELEASE_VERSION" || exit 1
+    inv -e ${INVOKE_TASK} --hardened-runtime --python-runtimes "$PYTHON_RUNTIMES" --major-version "$AGENT_MAJOR_VERSION" --release-version "$RELEASE_VERSION" || exit 1
```
Comment (CelianR) [inline scripts/build_script.sh:60]: NIT: ```suggestion
    inv -e $INVOKE_TASK --hardened-runtime --python-runtimes "$PYTHON_RUNTIMES" --major-version "$AGENT_MAJOR_VERSION" --release-version "$RELEASE_VERSION" || exit 1
```

File: scripts/build_script.sh [hunk 3]
```diff
@@ -48,13 +48,18 @@ if [ -n "$INTEGRATIONS_CORE_REF" ]; then
...
     # Lock the keychain once we're done
     security lock-keychain "$KEYCHAIN_NAME"
 else
-    inv -e agent.omnibus-build --skip-sign --python-runtimes "$PYTHON_RUNTIMES" --major-version "$AGENT_MAJOR_VERSION" --release-version "$RELEASE_VERSION" || exit 1
+    inv -e ${INVOKE_TASK} --skip-sign --python-runtimes "$PYTHON_RUNTIMES" --major-version "$AGENT_MAJOR_VERSION" --release-version "$RELEASE_VERSION" || exit 1
```
Comment (CelianR) [inline scripts/build_script.sh:64]: NIT: ```suggestion
    inv -e $INVOKE_TASK --skip-sign --python-runtimes "$PYTHON_RUNTIMES" --major-version "$AGENT_MAJOR_VERSION" --release-version "$RELEASE_VERSION" || exit 1
```

File: scripts/build_script.sh (hunk 2 above)
Comment (CelianR) [reply scripts/build_script.sh:60]: (or you can add `"$INVOKE_TASK"`)

File: scripts/build_script.sh (hunk 1 above)
Comment (chouquette) [reply scripts/build_script.sh:52]: Ah indeed, good point!

**Pull request merged**